### ISSUE TITLE
Remove deprecated API in t.names.root

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5564]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5564|
|Type|task|
|Created|2012-03-19 18:51:23Z|
|Branch|https://github.com/twisted/twisted/tree/names-root-5564|

The following were deprecated in 10.0 (r27740/[#10648](https://github.com/twisted/twisted/issues/10648)) and can be removed:

 - `retry`
 - `Resolver.discoveredAuthority`
 - `lookupNameservers`
 - `lookupAddress`
 - `extractAuthority`
 - `discoverAuthority`

Attachments:

* [5564.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/180/18072740660c6a233e06cc37e58f318dc2b1688b/f679914152bc38fd607e23364440d85544a9e988.patch) (12763 bytes) - added by piffey on 2012-03-20 14:47:31Z - 
* [5564.removal](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/180/18072740660c6a233e06cc37e58f318dc2b1688b/a76e9eaa6426cbe036fe8b9daf5e80d1de205334.removal) (681 bytes) - added by piffey on 2012-03-20 17:49:35Z - Must have forgotten to grab the new file with svn diff. Whoops!
* [5564.removal-2](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/180/18072740660c6a233e06cc37e58f318dc2b1688b/5ab68d7a5ff1e1ad3f07a5abb1faa4dd07e3bf3d.removal-2) (197 bytes) - added by piffey on 2012-03-20 18:53:20Z - Revision to cut down on repetition.

<details><summary>Searchable metadata</summary>

```
trac-id__5564 5564
type__task task
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_names_root_5564 branches/names-root-5564
branch_author__piffey piffey
status__closed closed
resolution__fixed fixed
component__names names
keywords__easy easy
time__1332183083000000 1332183083000000
changetime__1362000497000000 1362000497000000
version__None None
owner__tom_prince tom.prince
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="piffey's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|piffey commented|
|-|-|

First time contributing, followed the model of other tickets and hope everything is proper. Added news file already as well.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [33962]) Branching to 'names-root-5564'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set owner to piffey|
|-|-|

Replying to [piffey](#note_1):
> First time contributing, followed the model of other tickets and hope everything is proper. Added news file already as well.

Thanks. All looks good but the news file wasn't included in the patch. Can you attach it here?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Thanks.  The news file doesn't need to mention the removed unit tests, only the removed public APIs.  Also, if you can factor out the duplicate text in a way that doesn't make it sound awkward, that'd be great.

To future reviewers, though this looks pretty good and is probably just about in shape to merge to trunk, please do not merge to trunk yet.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Replying to [exarkun](#note_4):
> To future reviewers, though this looks pretty good and is probably just about in shape to merge to trunk, please do not merge to trunk yet.

I added the news file in r34122. Assigning to exarkun for review based on his comment 4.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

Don't think there's anything left to review here.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

Replying to [exarkun](#note_7):
> Don't think there's anything left to review here.

Alright, I'll merge it then.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `new`|
|-|-|

Please don't.  There's no reason to rush to remove deprecated things.  We should leave them *as long as possible*.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_9):
> Please don't.  There's no reason to rush to remove deprecated things.  We should leave them *as long as possible*.

Thanks for the clarification, comment 7 got me confused.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_11"></a>|@tomprince commented|
|-|-|

I'm working on [#6044](https://github.com/twisted/twisted/issues/6044), and so need to change all uses of `deferredGenerator`. `discoverAuthority` uses it, so this ticket can probably land now.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun set owner to @tomprince|
|-|-|

That's reasonable.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_13"></a>|@tomprince set status to `closed`|
|-|-|

(In [37287]) Merge names-root-5564: Remove deprecated API in twisted.names.root.

Author: piffey
Reviewers: thijs
Fixes: [#5564](https://github.com/twisted/twisted/issues/5564)

The following were deprecated in 10.0 (r27740/[#10648](https://github.com/twisted/twisted/issues/10648)) and can be removed:

 * retry
 * Resolver.discoveredAuthority
 * lookupNameservers
 * lookupAddress
 * extractAuthority
 * discoverAuthority
